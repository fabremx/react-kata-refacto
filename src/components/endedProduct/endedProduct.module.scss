@import "../../assets/styles/colors.scss";

$orange: #ec7a5c;

.photo div div {
  font-size: 11px;
  background-color: $orange;
  color: white;
  font-weight: 600;
  text-align: center;
  padding: 5px;
  border-radius: 3px;
}

.name span {
  color: $orange;
  font-weight: 600;
}
