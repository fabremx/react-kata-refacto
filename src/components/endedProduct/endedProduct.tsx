import { ReactElement } from "react";
import baseProductStyles from "../baseProduct/baseProduct.module.scss";
import styles from "./endedProduct.module.scss";

interface Props {
  image: string;
  title: string;
}

export function EndedProduct({ image, title }: Props): ReactElement {
  return (
    <>
      <td className={`${baseProductStyles.photo} ${styles.photo}`}>
        <div data-testid="photoEndedProduct">
          <img src={image} alt="product" />
          <div>End soon</div>
        </div>
      </td>

      <td
        className={`${baseProductStyles.name} ${styles.name}`}
        data-testid="titleEndedProduct"
      >
        {title} <span>(End Soon)</span>
      </td>
    </>
  );
}
