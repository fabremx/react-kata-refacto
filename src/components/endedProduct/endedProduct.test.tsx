import { render } from "@testing-library/react";
import { EndedProduct } from "./endedProduct";

describe("EndedProduct", () => {
  it("should render correctly the component", () => {
    // Given
    const tableRow = document.createElement("tr");
    // When
    const { asFragment } = render(
      <EndedProduct image={"Image"} title={"Title"} />,
      {
        container: document.body.appendChild(tableRow),
      }
    );
    // Then
    expect(asFragment()).toMatchSnapshot();
  });
});
