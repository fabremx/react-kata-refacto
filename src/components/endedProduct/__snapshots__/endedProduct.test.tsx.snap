// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndedProduct should render correctly the component 1`] = `
<DocumentFragment>
  <div
    data-testid="photoEndedProduct"
  >
    <img
      alt="product"
      src="Image"
    />
    <div>
      End soon
    </div>
  </div>
  Title 
  <span>
    (End Soon)
  </span>
</DocumentFragment>
`;
